ballerina/json_schema.bal

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Defines constraints for JSON schema string type.
+#
+# + maxLength - The maximum number of characters allowed in the string.
+# + minLength - The minimum number of characters required in the string.
+# + pattern - A regular expression that the string must match.
+# + format - The format that the string must adhere to (e.g., email, date-time).
+public type StringValidationRec record {|
+    int:Unsigned32 maxLength?;
+    int:Unsigned32 minLength?;
+    string:RegExp pattern?;
+    string format?;
+|};
+
+# The annotation is used to specify validation constraints for string type.
+public annotation StringValidationRec StringValidation on type;
+
+# Defines constraints for JSON schema number type.
+#
+# + maximum - The inclusive upper bound of the constrained type.
+# + minimum - The inclusive lower bound of the constrained type.
+# + exclusiveMaximum - The exclusive upper bound of the constrained type.
+# + exclusiveMinimum - The exclusive lower bound of the constrained type.
+# + multipleOf - The number must be a multiple of this value.
+public type NumberValidationRec record {|
+    int|float|decimal maximum?;
+    int|float|decimal minimum?;
+    int|float|decimal exclusiveMaximum?;
+    int|float|decimal exclusiveMinimum?;
+    int|float|decimal multipleOf?;
+|};
+
+# The annotation is used to specify validation constraints for number type.
+public annotation NumberValidationRec NumberValidation on type;
+
+# Defines constraints for JSON Schema array type.
+#
+# + minItems - The minimum number of items required in the array.
+# + maxItems - The maximum number of items allowed in the array.
+# + uniqueItems - Specifies whether all items in the array must be unique.
+# + unevaluatedItems - Specifies a schema for items that are not evaluated by contains.
+# + contains - Specifies a schema that a specific number of items in the array must match.
+public type ArrayValidationRec record {|
+    int:Unsigned32 minItems = 0;
+    int:Unsigned32 maxItems?;
+    boolean uniqueItems?;
+    record {|
+        int:Unsigned32 minContains = 0;
+        int:Unsigned32 maxContains?;
+        typedesc<json> contains;
```

Comment:
Shouldn't this be value or schema or something like that instead of repeating contains?